flocker/acceptance/test_api.py

```diff
@@ -17,10 +17,32 @@
 from pyrsistent import PRecord, field, CheckedPVector
 
 from ..testtools import loop_until
+from .testtools import (
+    MONGO_IMAGE, require_mongo, get_mongo_client,
+)
 from ..node.agents.test.test_blockdevice import REALISTIC_BLOCKDEVICE_SIZE
 from ..control.httpapi import REST_API_PORT
 
 
+def verify_socket(host, port):
+    """
+    Wait until the destionation can be connected to.
+
+    :param bytes host: Host to connect to.
+    :param int port: Port to connect to.
+
+    :return Deferred: Firing when connection is possible.
+    """
+    def can_connect():
+        s = socket.socket()
+        conn = s.connect_ex((host, port))
+        print "Connection status", conn
+        return False if conn else True
+
+    dl = loop_until(can_connect)
+    return dl
+
```

Comment:
Moving `verify_socket` is not strictly related to the issue or used in the "create container with dataset" test. Still, seems a small enough inline improvement that it's worth keeping in this branch.